aws-api/src/main/java/com/amplifyframework/api/aws/auth/AuthRuleProcessor.java

```diff
@@ -0,0 +1,222 @@
+                            "you did, check that the value you specified in your schema is present in the access key."
+            );
+        }
+
+        return identityValue;
+    }
+
+    private ArrayList<String> getUserGroups(AuthorizationType authType) throws ApiException {
+        // Custom groups claim isn't supported yet.
+        if (!AuthorizationType.AMAZON_COGNITO_USER_POOLS.equals(authType)) {
+            throw new ApiException("Custom groups claim is not supported yet.",
+                    "Please use Amazon Cognito User Pools to authorize your API.");
+        }
+
+        ArrayList<String> groups = new ArrayList<>();
+        final String GROUPS_KEY = "cognito:groups";
+
+        try {
+            JSONObject accessToken = CognitoJWTParser.getPayload(getAuthToken(authType));
+
+            if (accessToken.has(GROUPS_KEY)) {
+                JSONArray jsonGroups = accessToken.getJSONArray(GROUPS_KEY);
+
+                for (int i = 0; i < jsonGroups.length(); i++) {
+                    groups.add(jsonGroups.getString(i));
+                }
+            }
+        } catch (JSONException | CognitoParameterInvalidException error) {
+            throw new ApiException(
+                    "Failed to parse groups from auth rule.",
+                    error,
+                    "This should never happen - see attached exception for more details and report to us on GitHub."
+            );
+        }
+
+        return groups;
+    }
+
+    private String getAuthToken(AuthorizationType authType) throws ApiException {
+        switch (authType) {
+            case AMAZON_COGNITO_USER_POOLS:
+                CognitoUserPoolsAuthProvider cognitoProvider = authProvider.getCognitoUserPoolsAuthProvider();
+                if (cognitoProvider == null) {
+                    cognitoProvider = new DefaultCognitoUserPoolsAuthProvider();
+                }
+                return cognitoProvider.getLatestAuthToken();
+            case OPENID_CONNECT:
+                OidcAuthProvider oidcProvider = authProvider.getOidcAuthProvider();
+                if (oidcProvider == null) {
+                    throw new ApiException(
+                            "OidcAuthProvider interface is not implemented.",
+                            "Configure AWSApiPlugin with ApiAuthProviders containing an implementation of " +
+                                    "OidcAuthProvider interface that can vend a valid JWT token."
+                    );
+                }
+                return oidcProvider.getLatestAuthToken();
+            case API_KEY:
+            case AWS_IAM:
+            case NONE:
+            default:
+                throw new ApiException(
+                        "Failed to obtain access token from the configured auth provider.",
+                        "Verify that the API is configured with either Cognito User Pools or OpenID Connect."
```

Comment:
Should this say something about like `"@auth is only supported when suing Cognito User Pools or OpenID connect?"`